semgrep/semgrep/pattern_match.py

```diff
@@ -63,5 +63,23 @@ def end(self) -> Dict[str, Any]:
             del end["offset"]
         return end
 
+    def get_metavariable_value(self, metavariable: str) -> str:
+        """
+            Use metavars start and end to read into the file to find what the
+            metavariable in this pattern match maps to in the file
+
+            Assumes METAVARIABLE is a key in self.metavars
+        """
+        # Offsets are start inclusive and end exclusive
+        start_offset = self.metavars[metavariable].get("start").get("offset")
+        end_offset = self.metavars[metavariable].get("end").get("offset")
```

Comment:
If `start` or `end` are missing as keys, we'll fail anyway, so I'll update this to `["end"]["offset"]` to make the error message more clear.